tardis/energy_input/gamma_ray_packet_source.py

```diff
@@ -561,30 +585,40 @@ def create_packet_nus(
             DataFrame of packets
         positronium_fraction : float
             The fraction of positrons that form positronium
-        positronium_energy : array
-            Array of positronium frequency-energies to sample
-        positronium_intensity : array
-            Array of positronium intensities to sample
-
+            default is 0.0
         Returns
         -------
         array
             Array of sampled frequency-energies
         """
-        energy_array = np.zeros(no_of_packets)
-        zs = np.random.random(no_of_packets)
-        for i in range(no_of_packets):
-            # positron
-            if packets.iloc[i]["decay_type"] == "bp":
-                # positronium formation 75% of the time if fraction is 1
-                if zs[i] < positronium_fraction and np.random.random() < 0.75:
-                    energy_array[i] = sample_energy(
-                        positronium_energy, positronium_intensity
-                    )
-                else:
-                    energy_array[i] = 511
-            else:
-                energy_array[i] = packets.iloc[i]["radiation_energy_kev"]
+
+        energy_array = np.zeros(number_of_packets)
+
+        all_packets = np.array([True] * number_of_packets)
+
+        # positronium formation if fraction is greater than zero
+        positronium_formation = (
+            np.random.random(number_of_packets) < positronium_fraction
+        )
+        # annihilation line of positrons
+        annihilation_line = packets["radiation_energy_keV"] == 511.0
```

Comment:
Probably 511 should be a constant defined somewhere with a clear name like POSITRON_ANNIHILATION_ENERGY or something. Especially since it's used in a couple places.